### PR TITLE
refactor(query): migrate serializations to expression

### PR DESCRIPTION
**Pull request by b41sh**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

migrate serializations to expression

Fixes #issue

### CODE REVIEW

File: src/query/expression/src/types.rs [hunk 1]
```diff
@@ -104,6 +118,94 @@ impl DataType {
...
+        match self {
+            DataType::Null => {
+                let serializer = NullSerializer::try_create(column)?;
+                Ok(TypeSerializerImpl::Null(serializer))
+            }
+            DataType::Boolean => {
+                let serializer = BooleanSerializer::try_create(column)?;
+                Ok(TypeSerializerImpl::Boolean(serializer))
+            }
+            DataType::Number(num_ty) => match num_ty {
```
Comment (sundy-li) [inline src/query/expression/src/types.rs:132]: can use macro to fold the codes here

Comment (b41sh) [reply src/query/expression/src/types.rs:132]: good idea

**Comment by youngsofun** [comment]
FormatSettings and TypeSerializerImpl is changing in https://github.com/datafuselabs/databend/pull/8566,
and may need further change (need some disscution later).
what about only migrating `write_field` and `write_field_json` just to exploit how to integrate this framework to the new expression?

**Comment by b41sh** [comment]
> FormatSettings and TypeSerializerImpl is changing in #8566, and may need further change (need some disscution later). what about only migrating `write_field` and `write_field_json` just to exploit how to integrate this framework to the new expression?

Unneeded functions have been removed. PTAL @youngsofun

**Review by sundy-li — APPROVED** [review]

**Comment by BohuTANG** [comment]
@b41sh please rebase :)

**Review by youngsofun — APPROVED** [review]

**Pull request merged**